### PR TITLE
Daily notification and Widget

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 75432: Notification and Widget](https://edu-redmine.sun-asterisk.vn/issues/75432)
## WHAT
- add Daily notification
- add Widget
## Evidence (Screenshot or Video)

https://github.com/awesome-academy/HCM_Android_2024_Weather_App_PJ2/assets/159242052/6694b024-0902-4b45-8986-7ac7c0585cc8

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/blob/master/eng/android/coding_convention_android_kotlin.md | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(Other notes)*

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: app/src/main/java/com/example/weather/notification/DailyWorker.kt [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                    applicationContext.getString(R.string.temperature)
+                    } $temperature℃        ${
+                    applicationContext.getString(R.string.wind_speed)
+                    } $windSpeed km/h"
+                showNotification(notificationTitle, notificationContent)
+            }
+        }
+    }
+
+    @SuppressLint("MissingPermission")
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/notification/DailyWorker.kt:96]:
```suggestion
    // FIXME Update later
    @SuppressLint("MissingPermission")
```
Ca này gấp gáp nên anh merged nhé, em add tag này vào sau phải fix không chơi như này

File: app/src/main/java/com/example/weather/notification/DailyWorker.kt [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import java.io.IOException
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+class DailyWorker(context: Context, workerParameters: WorkerParameters) :
+    Worker(context, workerParameters), KoinComponent {
+
+    @SuppressLint("MissingPermission")
```
Comment (daolq-2712) [inline app/src/main/java/com/example/weather/notification/DailyWorker.kt:37]: tương tự bên trên

**Review by daolq-2712 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/weather/notification/DailyWorker.kt (hunk 1 above)
Comment (ghost) [reply app/src/main/java/com/example/weather/notification/DailyWorker.kt:96]: @daolq-2712 dạ lát chiều em sẽ fix luôn ạ.

**Review by daolq-2712 — APPROVED** [review]
OK!

**Pull request merged**